### PR TITLE
[NOD-648] Add TestProcessDelayedBlocks

### PR DESCRIPTION
**Pull request by someone235**

### CODE REVIEW

File: blockdag/process_test.go [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
 import (
+	"github.com/kaspanet/kaspad/util"
 	"path/filepath"
 	"testing"
+	"time"
 
 	"github.com/kaspanet/kaspad/dagconfig"
 	"github.com/kaspanet/kaspad/util/daghash"
+	_ "net/http/pprof"
```
Comment (svarogg) [inline blockdag/process_test.go]: Please delete

File: blockdag/process_test.go [hunk 2]
```diff
@@ -69,3 +71,167 @@ func TestProcessOrphans(t *testing.T) {
...
+	if isOrphan {
+		t.Fatalf("ProcessBlock incorrectly returned delayedBlock " +
+			"is an orphan\n")
+	}
+	if !isDelayed {
+		t.Fatalf("ProcessBlock incorrectly returned delayedBlock " +
+			"is not delayed\n")
+	}
+
+	if dag2.BlockExists(delayedBlock.BlockHash()) {
```
Comment (stasatdaglabs) [inline blockdag/process_test.go]: BlockExists and HaveBlock sounds too similar to have opposite behaviors. Please consider renaming either one or both of them.

File: blockdag/process_test.go [hunk 3]
```diff
@@ -69,3 +71,167 @@ func TestProcessOrphans(t *testing.T) {
...
+	}
+	if dag2.BlockExists(delayedBlockChild.BlockHash()) {
+		t.Errorf("delayedBlockChild shouldn't be added to the DAG because its parent is not in the DAG")
+	}
+
+	secondsUntilDelayedBlockIsValid := delayedBlock.Header.Timestamp.Unix() - int64(dag2.TimestampDeviationTolerance) - dag2.AdjustedTime().Unix() + 1
+	if secondsUntilDelayedBlockIsValid < 0 {
+		t.Fatalf("the test took too much time, so delayedBlock timestamp is accidently valid now")
+	}
+	time.Sleep(time.Duration(secondsUntilDelayedBlockIsValid) * time.Second)
```
Comment (stasatdaglabs) [inline blockdag/process_test.go]: Please don't use time.Sleep inside tests. Manipulate timeSource instead.

Comment (svarogg) [reply blockdag/process_test.go]: In addition - add comment explaining that now we moved the clock to a time where all blocks can be accepted

File: blockdag/blockindex.go [hunk 4]
```diff
@@ -38,7 +38,7 @@ func newBlockIndex(db database.DB, dagParams *dagconfig.Params) *blockIndex {
 	}
 }
 
-// HaveBlock returns whether or not the block index contains the provided hash.
+// IsKnownBlock returns whether or not the block index contains the provided hash.
 //
 // This function is safe for concurrent access.
 func (bi *blockIndex) HaveBlock(hash *daghash.Hash) bool {
```
Comment (stasatdaglabs) [inline blockdag/blockindex.go]: Please rename this to IsKnownBlock.

File: peer/peer.go [hunk 5]
```diff
@@ -198,7 +198,7 @@ type Config struct {
 	// to the peer as needed.
 	SelectedTipHash func() *daghash.Hash
 
-	// BlockExists determines whether a block with the given hash exists in
+	// IsInDAG determines whether a block with the given hash exists in
 	// the DAG.
 	BlockExists func(*daghash.Hash) bool
```
Comment (stasatdaglabs) [inline peer/peer.go]: Please rename this to IsInDAG.

File: blockdag/blockindex.go (hunk 4 above)
Comment (someone235) [reply blockdag/blockindex.go]: Actually here it should be HaveBlock

Comment (stasatdaglabs) [reply blockdag/blockindex.go]: :+1:

**Pull request merged**